universe/ValueRefs.h

```diff
@@ -176,6 +176,33 @@ struct FO_COMMON_API Statistic final : public Variable<T>
     std::unique_ptr<ValueRef<V>>          m_value_ref;
 };
 
+/** The variable TotalFighterShots class.   The value returned by this node is
+  * computed from the gamestate; the number of shots of a launched fighters
+  * of the source carrier is counted (and added up) for all combat bouts
```

Comment:
is combat bout limits the only use of the sampling condition?

Reply:
> for fighters i implemented both a TotalFighterShots condition = OPTIONAL_TARGET_FILTER condition (necessary to use if one uses CombatBout) and a fallback which uses the hangar part combatTargets if such exists (so the current hangar content is fine without using TotalFighterShots).

Combat bout limits is the only use case which is not handled by the fallback (using combatTargets). Mostly for performance concerns I did not want to eval the target condition for every bout in all cases but only once per carrier per combat. (The performance could be also gained by introducing BoutInvariance for valuerefs).

Also see [comment above](https://github.com/freeorion/freeorion/pull/3356#issuecomment-805274270)

Reply:
Note this would be the FOCS version when not using the fallback for strikers
```
totalFighterDamageEstimation = TotalFighterShots condition = All
totalShipDamageEstimation = TotalFighterShots condition = All * ShipPartMeter part = "FT_HANGAR_3" meter = Capacity object = Source.ID
```